contracd/src/contracd.cpp

```diff
@@ -47,8 +47,9 @@ int main(int argc, char *argv[])
         setlinebuf(stdout);
         setlinebuf(stderr);
 
-        dbus = new DBusInterface();
-        result = dbus == nullptr ? 0 : 1;
+        bool dbusInterfaceResult;
+        dbus = new DBusInterface(nullptr, &dbusInterfaceResult);
+        result = (dbus == nullptr || !dbusInterfaceResult) ? -1 : 0;
```

Comment:
You can pass `result` in directly, no need for the additional `dbusInterfaceResult`.

I think I'd also be inclined to drop the null pointer check (I suspect I originally included it by mistake). But by all means keep it if you think it's useful though.